app/account/profile/page.tsx

```diff
@@ -0,0 +1,272 @@
+'use client'
+
+import { Avatar, AvatarFallback, AvatarImage } from '@/components/ui/avatar'
+import React from 'react'
+import UseProfile from '@/assets/image/user-profile.png'
+import { Button } from '@/components/ui/button'
+import {
+    Dialog,
+    DialogContent,
+    DialogHeader,
+    DialogTitle,
+    DialogTrigger,
+} from '@/components/ui/dialog'
+import {
+    Form,
+    FormControl,
+    FormField,
+    FormItem,
+    FormMessage,
+} from '@/components/ui/form'
+import { useForm } from 'react-hook-form'
+import { z } from 'zod'
+import { zodResolver } from '@hookform/resolvers/zod'
+import { PasswordResetSchema } from '@/lib/utils/schema'
+import { toast } from '@/components/ui/use-toast'
+import { Input } from '@/components/ui/input'
+import { CameraIcon } from 'lucide-react'
+
+type UserDataType = {
+    userName: string
+    email: string
+}
+
+function page() {
+    const [isEditing, setIsEditing] = React.useState<boolean>(false)
+    const [isEditingUsername, setIsEditingUsername] =
+        React.useState<boolean>(false)
+    const [isEditingUserEmail, setIsEditingUserEmail] =
+        React.useState<boolean>(false)
+    const [isClient, setIsClient] = React.useState<boolean>(false)
+
+    const [userData, setUserData] = React.useState<UserDataType>({
+        userName: '@leonardodiCaprio',
+        email: 'leonardodiCaprio@gmail.com',
+    })
+
+    const form = useForm<z.infer<typeof PasswordResetSchema>>({
+        resolver: zodResolver(PasswordResetSchema),
+        defaultValues: {
+            currentPassword: '',
+            newPassword: '',
+            passwordConfirm: '',
+        },
+    })
+
+    const handleUserNameChange = (evt: React.ChangeEvent<HTMLInputElement>) => {
+        setUserData((prev) => ({
+            ...prev,
+            userName: evt.target.value,
+        }))
+    }
+
+    const handleEmailChange = (evt: React.ChangeEvent<HTMLInputElement>) => {
+        setUserData((prev) => ({
+            ...prev,
+            email: evt.target.value,
+        }))
+    }
+
+    const onSubmit = async () => {
+        form.reset()
+        toast({
+            title: 'Password Reset',
+            description: 'Password reset successfully',
+        })
+    }
+
+    React.useEffect(() => {
```

Comment:
Looks great otherwise, but wondering about this.

Reply:
Importing the hooks from react would be preferred @yaredow

Reply:
> Looks great otherwise, but wondering about this.

I don't get it. Why I used a useEffect?

Reply:
> Importing the hooks from react would be preferred @yaredow 

I'm pretty sure I didn't import them from Angular. Lol 

There is no difference in functionality between them. I just prefer using the hooks like this. I can change it though.

Reply:
Yes, it might be tidier and easier for you to just import useEffect, but don't worry about it. I was wondering more about whether we could avoid the hydration errors without using an effect hook to determine component client side load.

Let's merge this in for now, but you might notice a slight performance delay with the header. We can address this in the future - might be better to just move ahead, we can create another ticket for nice tweaks like this.

Well done, it's a good job!